src/lib/Parser.hs

```diff
@@ -750,6 +750,7 @@ ops =
   [ [InfixL $ sym "." $> mkGenApp TabArrow, symOp "!"]
   , [InfixL $ sc $> mkApp]
   , [prefixNegOp]
+  , [annotatedExpr]
```

Comment:
I always find myself wondering how strongly does the type annotation operator bind. Do you think it would maybe make sense to enforce people to always wrap those annotations in parens? Otherwise it's super unclear whether the type annotation in `arr.i:type` applies to the index or the array element.

Reply:
I guess to avoid all ambiguity you'd need two sets of parens, right? As in `f ((arr.i) : type)`? Otherwise you still need to define its precedence with respect to other operators.

Reply:
I suppose you could force the user to be explicit by making `:` have the lowest possible precedence but then throwing an error afterward if it looks ambiguous (i.e. try to take as much as possible, but then error if you were able to take more than a single leaf expression)?

Reply:
Yeah I agree it might be tricky. I'm not sure if the heuristic you proposed would work, since sometimes you do want to have an application on the right. That's the basic case of `Fin n`.

Reply:
I think I'll leave it as-is for now, then.